pkg/gardenlet/controller/federatedseed/network-policy/controller.go

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package network_policy
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/controllerutils"
+	"github.com/gardener/gardener/pkg/gardenlet/controller/federatedseed/network-policy/helper"
+
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	kubeinformers "k8s.io/client-go/informers"
+	kubecorev1listers "k8s.io/client-go/listers/core/v1"
+	kubenetworkingv1listers "k8s.io/client-go/listers/networking/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// Controller watching the endpoints resource "kubernetes" of the Seeds's kube-apiserver in the default namespace
+// to keep the NetworkPolicy "allow-to-seed-apiserver" in sync.
+type Controller struct {
+	log      *logrus.Entry
+	recorder record.EventRecorder
+
+	namespaceReconciler reconcile.Reconciler
+
+	endpointSynced  cache.InformerSynced
+	endpointsLister kubecorev1listers.EndpointsLister
+
+	networkPoliciesSynced cache.InformerSynced
+	networkPoliciesLister kubenetworkingv1listers.NetworkPolicyLister
+
+	namespaceSynced cache.InformerSynced
+	namespaceQueue  workqueue.RateLimitingInterface
+	namespaceLister kubecorev1listers.NamespaceLister
+
+	workerCh               chan int
+	numberOfRunningWorkers int
+	waitGroup              sync.WaitGroup
+}
+
+var (
+	ShootNamespaceSelector = labels.SelectorFromSet(labels.Set{
```

Comment:
doc strings missing for exported variables

Reply:
Why this needs to be a global variable in the first place? Put in the `Controller` struct?